File: 2520-count-the-digits-that-divide-a-number/2520-count-the-digits-that-divide-a-number.cpp
class Solution {
public:
    int countDigits(int num) {
        int counter = 0;
        int n = num;
        while (n > 0){
            int dig = n % 10;
            if(dig != 0 && num % dig == 0){
                counter++;
            }
           n = n / 10;
        }
         return counter;
    }
};

File: 2520-count-the-digits-that-divide-a-number/2520-count-the-digits-that-divide-a-number.py
class Solution:
    def countDigits(self, num: int) -> int:
        count = 0
        n = num

        while n > 0:
            r = n % 10

            if r != 0 and num % r == 0:
                count += 1

            n //= 10

        return count